src/libstd/sys/windows/backtrace.rs

```diff
@@ -157,3 +160,37 @@ unsafe fn _write(w: &mut Write) -> io::Result<()> {
 
     Ok(())
 }
+
+fn query_full_process_image_name() -> io::Result<PathBuf> {
+    super::fill_utf16_buf(|buf, mut sz| unsafe {
+        c::QueryFullProcessImageNameW(c::GetCurrentProcess(), 0, buf, &mut sz); sz
```

Comment:
Could you use `OpenProcess(PROCESS_QUERY_INFORMATION, FALSE, GetCurrentProcessId())` instead of `c::GetCurrentProcess()` like the https://github.com/rust-lang/rust/pull/37359 does?
`QueryFullProcessImageNameW` didn't work with pseudo-handle `-1` returned by `GetCurrentProcess` the last time I tried.